tasks/kernel_matrix_testing/compiler.py

```diff
@@ -32,6 +33,21 @@ def get_build_image_suffix_and_version() -> tuple[str, str]:
     return ci_vars['DATADOG_AGENT_BUILDIMAGES_SUFFIX'], ci_vars['DATADOG_AGENT_BUILDIMAGES']
 
 
+def has_ddtool_helpers() -> bool:
+    docker_config = Path("~/.docker/config.json").expanduser()
+    if not docker_config.exists():
+        return False
+
+    try:
+        with open(docker_config) as f:
+            config = json.load(f)
+    except json.JSONDecodeError:
+        # Invalid JSON (or empty file), we don't have the helper
+        return False
+
+    return "ddtool" in config.get("credHelpers", {}).values()
```

Comment:
FWIW, the ECR registries use `ecr-login` as the cred helper.

Reply:
Added that too for the detection. For me  `ddtool auth helpers install` only added the `ddtool` helpers but maybe that's changed recently.